### PR TITLE
Skip CNV/ODF upgrade when Cluster is already upgraded

### PR DESCRIPTION
**Pull request by ebattat**
## Type of change
Note: Fill **x** in []
- [x] bug
- [ ] enhancement
- [ ] documentation
- [ ] dependencies

## Description

Skip CNV/ODF upgrade when Cluster is already upgraded

### CODE REVIEW

File: benchmark_runner/common/clouds/BareMetal/bare_metal_operations.py [hunk 1]
```diff
@@ -239,6 +239,7 @@ def run_ocp_upgrade(self, oc: OC):
         """
         if f'Cluster version is {self._upgrade_ocp_version}' == oc.get_cluster_status():
             logger.info(f'Cluster is already upgraded to: {self._upgrade_ocp_version}')
+            return 'Cluster is already upgraded'
```
Comment (RobertKrawitz) [inline benchmark_runner/common/clouds/BareMetal/bare_metal_operations.py:242]: Should return True, not a string (or you need to change the comment above)..  Also note that there's no return if the version is not correct (existing code, but it looks wrong).

File: benchmark_runner/main/main.py [hunk 2]
```diff
@@ -159,10 +159,10 @@ def upgrade_ocp_bare_metal(step: str):
     bare_metal_operations.connect_to_provisioner()
     oc = bare_metal_operations.oc_login()
     if step == 'run_bare_metal_ocp_upgrade':
-        bare_metal_operations.run_ocp_upgrade(oc)
-        # The LSO/ODF upgrade must be run manually after the OCP upgrade for the channel version; it won't upgrade automatically.
-        bare_metal_operations.install_ocp_resources(resources=['lso'], upgrade_version=lso_version)
-        bare_metal_operations.install_ocp_resources(resources=['odf'], upgrade_version=odf_version)
+        if not bare_metal_operations.run_ocp_upgrade(oc) == 'Cluster is already upgraded':
```
Comment (RobertKrawitz) [inline benchmark_runner/main/main.py:162]: Is there anything you need to do if the status is anything different (e. g. if either the upgrade fails or the upgrade does something, so it's not previously upgraded)?

**Review by RobertKrawitz — CHANGES_REQUESTED** [review]

File: benchmark_runner/main/main.py (hunk 2 above)
Comment (ebattat) [reply benchmark_runner/main/main.py]: Add dedicate method is_ocp_already_upgraded

**Comment by ebattat** [comment]
@RobertKrawitz, any more comments ?

**Review by RobertKrawitz — APPROVED** [review]

**Pull request merged**